File: backend/app/schemas/hospital.py
from datetime import datetime
from typing import Optional, List
from decimal import Decimal
from pydantic import BaseModel, Field
from app.schemas.resource import ResourceCreate, ResourceResponse

class HospitalBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=255, description="Hospital name (required)")
    address: Optional[str] = Field(default=None, max_length=500)
    latitude: Optional[Decimal] = Field(default=None, ge=-90, le=90)
    longitude: Optional[Decimal] = Field(default=None, ge=-180, le=180)
    emergency_level: Optional[str] = Field(default="Level 1", max_length=50)
    trauma_center: bool = Field(default=False)
    status: str = Field(default="Active", max_length=50)

class HospitalCreate(HospitalBase):
    resources: Optional[List[ResourceCreate]] = Field(default_factory=list)

class HospitalUpdate(BaseModel):
    name: Optional[str] = Field(default=None, min_length=1, max_length=255)
    address: Optional[str] = Field(default=None, max_length=500)
    latitude: Optional[Decimal] = Field(default=None, ge=-90, le=90)
    longitude: Optional[Decimal] = Field(default=None, ge=-180, le=180)
    emergency_level: Optional[str] = Field(default=None, max_length=50)
    trauma_center: Optional[bool] = Field(default=None)
    status: Optional[str] = Field(default=None, max_length=50)

class HospitalResponse(HospitalBase):
    id: int
    created_at: datetime
    updated_at: datetime
    resources: List[ResourceResponse] = []

    class Config:
        from_attributes = True

File: frontend/src/types/hospital.ts
export type ResourceType =
  | 'ICU_BED'
  | 'GENERAL_BED'
  | 'VENTILATOR'
  | 'OXYGEN_BED'
  | 'TRAUMA_BED'
  | 'OPERATING_ROOM';

export interface HospitalResource {
  id: number;
  hospital_id: number;
  resource_type: ResourceType;
  total: number;
  available: number;
  reserved: number;
  last_updated: string;
  created_at: string;
  updated_at: string;
}

export interface Hospital {
  id: number;
  name: string;
  address?: string;
  latitude?: number;
  longitude?: number;
  emergency_level?: string;
  trauma_center: boolean;
  status: string;
  created_at: string;
  updated_at: string;
  resources: HospitalResource[];
}

export interface HospitalCreate {
  name: string;
  address?: string;
  latitude?: number;
  longitude?: number;
  emergency_level?: string;
  trauma_center?: boolean;
  status?: string;
}
